
### Exercise 3
A rental car company claims the average time to rent a car on its website is 60 seconds with a standard deviation of 30 seconds. A random sample of 36 customers attempted to rent a car on the website. 

1. If the average time to rent was 65 seconds. Is this enough evidence to contradict the company's claim at a 5% confidence level? What's the power?

2. If the analytics team believe the previous claim is incorrect and the average time to rent is 67 seconds, what is the sample size needed to design a sufficient test with 85% of power to test this new claim?


In [13]:
#1. If the average time to rent was 65 seconds. Is this enough evidence to contradict the company's claim at a 5% confidence level? What's the power?
import numpy as np
from scipy.stats import norm
import scipy.stats as st
import numpy as np
z=(65-60)/(30/np.sqrt(36))
norm.cdf(z, loc=0, scale=1)

#Calculate Power 
#Beta level: the probability of failing to reject the null hypothesis when it actually should be rejected (if real mean is 65) 
#Power: the probability of reject the null hypothesis when it actually should be rejected (if real mean is 65)

#Calculating the required sample mean at 95% cutoff for null hypothesis 
x_bar=1.645*(30/np.sqrt(36))+60
x_bar 

#Beta
z2=(68.225-65)/(30/np.sqrt(36))
beta=1-norm.cdf(z2, loc=0, scale=1)

#Power
power=1-beta
power


0.7405364071452198

The average time 65 second is not enough evidence to contradict the company's claim at a 5% confidence level. The power is 0.741. 

In [17]:
# Importing the library
import scipy.stats
  
# Determine the z-critical value
scipy.stats.norm.ppf(0.15)

-1.0364333894937898

#2. If the analytics team believe the previous claim is incorrect and the average time to rent is 67 seconds, what is the sample size needed to design a sufficient test with 85% of power to test this new claim?

Need to solve the following system of equations
Calculate the cutoff value for sample mean at the rejection region under null hypothesis.

**x_bar=60+1.645*30/np.sqrt(n)**

The power (The probability of reject the null) when it should be rejected is 85%, and the beta
is 15%. 
**(x_bar-67)/(30/np.sqrt(n))=-1.037**

The approximate sample size is n=132. 



